docs/candid/01-style-guide.md

```diff
@@ -2,3 +2,306 @@
 title: Candid Style Guide
 description: A style guide for writing Candid interfaces for Internet Computer canisters.
 ---
+
+## Candid
+
+Candid is a serialization format for the Internet Computer. It is used to define the interface of canisters, and to encode and decode messages that are sent to and from canisters.
+
+## Motivation
+
+Canisters are written in a variety of languages, as are the clients that interact with canisters, while Candid is designed to be language agnostic. This means that Candid interfaces should follow a consistent pattern that is not biased by the language used to write the canister interfaces it is describing.
+
+## Naming
+
+### Types
+
+Use `PascalCase` (upper camel case) for type definitions.
+
+This has no additional benefit aside from consistency.
+
+:::danger[**Don't**]
+
+**Don't** use inconsisent or alternative casings for type definitions.
+
+```protobuf
+type USER = record {};
+
+type post = record {};
+```
+
+:::
+
+:::tip[**Do**]
+
+**Do** use `PascalCase` casing for type definitions consisently.
+
+```protobuf
+type User = record {};
+
+type Post = record {};
+```
+
+:::
+
+### Record field names
+
+Use `lower_snake_case` for field definitions.
+
+This has no additional benefit aside from consistency.
+
+:::danger[**Don't**]
+
+**Don't** use inconsisent or alternative casings for field definitions.
+
+```protobuf
+type User = record {
+  Id : nat64;
+  displayName : text;
+};
+
+type Post = record {
+  ID : nat64;
+  USER_ID : nat64;
+  Content : text;
+};
+```
+
+:::
+
+:::tip[**Do**]
+
+**Do** use `lower_snake_case` casing for field definitions consisently.
+
+```protobuf
+type User = record {
+  user_id : nat64;
+  name : text;
+};
+
+type Post = record {
+  id : nat64;
+  user_id : nat64;
+  content : text;
+};
+```
+
+:::
+
+### Variants
+
+Use `lower_snake_case` for variant definitions.
+
+This has no additional benefit aside from consistency.
+
+:::danger[**Don't**]
+
+**Don't** use inconsisent or alternative casings for variant definitions.
+
+```protobuf
+type UserRole = variant {
+  Admin;
+  MODERATOR;
+  user;
+};
+```
+
+:::
+
+:::tip[**Do**]
+
+**Do** use `lower_snake_case` casing for variant definitions consisently.
+
+```protobuf
+type UserRole = variant {
```

Comment:
this example does not really contain lower_snake_case names, just lower case. Would make sense to add at least one with "_" in it